### DISCUSSION TITLE
Xamarin Toolkit Helpers

### DISCUSSION THREAD

**Discussion by SirJohnK** [CommunityToolkit/Maui#470] [General]
Will any of the Helpers be migrated from the Xamarin Community Toolkit?

Mostly thinking of:

- LocalizationResourceManager with extensions etc.
- WeakEventManager

Thank you for great work! (Will try to contribute soon)

**Comment by TheCodeTraveler** [comment]
Hi @SirJohnK,

Short answer, no. 

Longer answer, kinda. 

We promoted WeakEventManager to the official .NET MAUI SDK. It's no longer necessary for this Toolkit. 

And we are looking into adding `LocalizationResourceManager` to `CommunityToolkit.Common` since it isn't specific to .NET MAUI. 

You can find more details here:

- `DelegateWeakEventManager` https://github.com/CommunityToolkit/Maui/issues/92
- `WeakEventManager<T>` https://github.com/CommunityToolkit/Maui/issues/94
- `LocalizationResourceManager`  https://github.com/CommunityToolkit/Maui/issues/6

**Reply by SirJohnK** [reply]
Sounds great! Thank you for clearifying! 👍

**Comment by CliffAgius** [comment]
Closed as answered
